### ISSUE TITLE
Crash opening local RLMRealm with `dynamic` setting (to be copied to a synced Realm)

### ISSUE THREAD

**Issue by i-schuetz** [realm/realm-cocoa#5510]
## Goals
Open local Realm, in order to be copied to a synced Realm, in Swift.

## Expected Results
The local Realm can be opened and copied to a synced realm. After doing this, also, the state of the local database should be reflected in the remote realm / realm platform.

It should be also possible to do this while the local realm is open, i.e. there are active `Object`s in memory.

It would be great to correct the project which is linked at the bottom to do this, so we have a complete example for uploading the local state of a Swift offline-first app, for other users. There's as good as no documentation for this at the moment.

## Actual Results
Crash when opening local Realm.

> 'Realm at path  #'/Users/.../Library/Developer/CoreSimulator/Devices/C6268DC5-0CCC-471D-B74D-B6E26B7B8848/data/Containers/Data/Application/39AF1EC3-D996-4450-BF68-3205B4939F06/Documents/default.realm' already opened with different dynamic settings'

## Steps to Reproduce
I followed the steps/code from https://github.com/realm/realm-cocoa/issues/5381

The problem apparently is that my local realm is already open (I use it everywhere in the app - it's an offline first app) and not `dynamic`, while the local realm created in order to be copied is `dynamic`. My default local realm is `Realm` (not `RLMRealm`) instance, which doesn't seem to have a way to set `dynamic` on it.

If I leave `dynamic` out, it opens but crashes later while copying, in table.hpp `inline void Table::bump_version(bool bump_global) const noexcept` with EXC_BAD_ACCESS (crash shows at the method's signature). Since `dynamic` is part of the recipe provided in https://github.com/realm/realm-cocoa/issues/5381 I assume it's necessary and could be related with this crash.

## Version of Realm and Tooling

Realm framework version: 3.0.2

Realm Object Server version: 2.1.1

Xcode version: 9.2

iOS/OSX version: 11.2 (Simulator)

Dependency manager + version: CocoaPods 1.3.1

### Additional question

Is there hope for this process to be more straightforward in the future? Having to use `RLMRealm`, with properties from `Realm.Private`, points to a missing api (and the fact that for Swift it's also not documented).

## Edit: Added link to test project

I created a project to test the described issues: https://github.com/i-schuetz/convert-realm
It has buttons to register and login to a local realm and create some dummy data, so everything can be tested.

Notes:

- When there's no local Realm open, opening a Realm with `dynamic` works (in my example this means no `things` instance variable. (In my real project it would be difficult to detach all the realm objects before syncing though.)

- When there's a local Realm open, it only works commenting the line where `dynamic` is set. Here I wonder: What's this setting for?

- In both cases (when it works), I still don't see any data in the Realm platform browser after copy completion (see screenshot). What else do I have to do?

<img width="798" alt="screen shot 2017-12-09 at 18 08 42" src="https://user-images.githubusercontent.com/1381744/33800614-bd854e3c-dd43-11e7-85d7-b393fcf02b54.png">

**Comment by i-schuetz** [comment]
I created a project to test the described issues: https://github.com/i-schuetz/convert-realm

Notes:

- When there's no local Realm open, opening a Realm with `dynamic` works (in my example this means no `things` instance variable. In my real project it would be difficult to detach all the realm objects before syncing.

- When there's a local Realm open, it only works commenting `dynamic`. Apparently everything still works. Here I wonder: What's this setting for?

- In both cases (when it works), I still don't see any data in the Realm platform browser after login / the copy is complete (see screenshot)
<img width="798" alt="screen shot 2017-12-09 at 18 08 42" src="https://user-images.githubusercontent.com/1381744/33797616-91d237a6-dd0b-11e7-9a27-830720beb493.png">
. Do I have to do something else to upload it?

**Comment by i-schuetz** [comment]
Is there anything that I can improve in this issue to get feedback? Not sure if the problem I'm describing is unclear, or what else may be preventing an answer. Happy to correct it. I also added an example project.

**Comment by i-schuetz** [comment]
Update: I managed to solve most of the issues, but a new crash appeared, this time during the migration - EXC_BAD_ACCESS in a `Table::bump_version(bool bump_global)`. It seems to happen only for specific objects (this was in my real project) as I was able to do the migration successfully for the small example project I linked in this issue.

Anyway I ended copying my objects one by one, via `Realm.create`. This was tedious as I have a lot of objects, with many relationships between them and lists but it works.
